src/runner.rs

```diff
@@ -0,0 +1,174 @@
+use std::path::{Path, PathBuf};
+
+use crate::context::Context;
+use crate::error::IxaError;
+use crate::global_properties::ContextGlobalPropertiesExt;
+use crate::random::ContextRandomExt;
+use crate::report::ContextReportExt;
+use clap::{Args, Command, FromArgMatches as _};
+
+#[derive(Args, Debug)]
+pub struct BaseArgs {
+    /// Random seed
+    #[arg(short, long, default_value = "0")]
+    pub seed: u64,
+
+    /// Optional path for a global properties config file
+    #[arg(short, long, default_value = "")]
+    pub config: String,
+
+    /// Optional path for report output
+    #[arg(short, long, default_value = "")]
+    pub output_dir: String,
+}
+
+#[derive(Args)]
+pub struct PlaceholderCustom {}
+
+#[allow(clippy::missing_errors_doc)]
+pub fn run_with_custom_args<A, F>(load: F) -> Result<(), Box<dyn std::error::Error>>
+where
+    A: Args,
+    F: Fn(&mut Context, BaseArgs, Option<A>) -> Result<(), IxaError>,
+{
+    let cli = Command::new("Ixa");
+    let cli = BaseArgs::augment_args(cli);
+    let cli = A::augment_args(cli);
+    let matches = cli.get_matches();
+
+    let base_args_matches = BaseArgs::from_arg_matches(&matches)?;
+    let custom_matches = A::from_arg_matches(&matches)?;
+
+    run_with_args_internal(base_args_matches, Some(custom_matches), load)
+}
+
+#[allow(clippy::missing_errors_doc)]
+pub fn run_with_args<F>(load: F) -> Result<(), Box<dyn std::error::Error>>
+where
+    F: Fn(&mut Context, BaseArgs, Option<PlaceholderCustom>) -> Result<(), IxaError>,
```

Comment:
Can you just do Option<()>. It doesnt' seem like there are any trait bounds.

Reply:
I have `A` implementing `Args` but i guess for this function it's not actually necessary